indico/modules/categories/controllers/management.py

```diff
@@ -16,10 +16,101 @@
 
 from __future__ import unicode_literals
 
+import os
+from io import BytesIO
+
+from flask import flash, redirect, request, session
+from PIL import Image
+
+from indico.modules.categories import logger
 from indico.modules.categories.controllers.base import RHManageCategoryBase
+from indico.modules.categories.forms import CategoryIconForm, CategorySettingsForm
 from indico.modules.categories.views import WPCategoryManagement
+from indico.util.fs import secure_filename
+from indico.util.i18n import _
+from indico.util.string import crc32
+from indico.web.flask.util import url_for
+from indico.web.forms.base import FormDefaults
+from indico.web.util import jsonify_data
+
+
+CATEGORY_ICON_DIMENSIONS = (16, 16)
+
+
+def _icon_data(category):
+    return {
+        'url': category.icon_url,
+        'filename': category.icon_metadata['filename'],
+        'size': category.icon_metadata['size'],
+        'content_type': category.icon_metadata['content_type']
+    }
 
 
 class RHManageCategoryContent(RHManageCategoryBase):
     def _process(self):
         return WPCategoryManagement.render_template('management/content.html', self.category, 'content')
+
+
+class RHManageCategorySettings(RHManageCategoryBase):
+    def _process(self):
+        defaults = FormDefaults(self.category,
+                                meeting_theme=self.category.default_event_themes['meeting'],
+                                lecture_theme=self.category.default_event_themes['lecture'])
+        form = CategorySettingsForm(obj=defaults, category=self.category)
+        icon_form = CategoryIconForm(obj=FormDefaults(self.category))
```

Comment:
`obj=self.category` works fine if you don't need any of the fanciness from FormDefaults